syntax/verilog_systemverilog.vim

```diff
@@ -289,7 +289,7 @@ endif
 
 if index(s:verilog_syntax_fold, "instance") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
     syn region verilogFold
-        \ start="^\s*\w\+\(\s*#\s*(\|\s\+\w\+\s*(\)\(.*)\s*\w\+\s*;\)\@!"
+        \ start="^\s*\(\<\(task\|function\|checker\)\>\)\@!\w\+\(\s*#\s*(\(.*)\s*\w\+\s*;\)\@!\|\s\+\(\<if\>\)\@!\w\+\s*(\)"
```

Comment:
Can you not re-arrange the ordering of the syntax expressions to achieve the same effect? Even though you've mentioned that this doesn't cover other scenarios, I don't think you will find that this will scale. In my experimental repo, I've made excessive use of the syntax priority ordering to minimize regular expressions.

`:help :syn-priority`